lapis2/src/main/kotlin/org/genspectrum/lapis/controller/DataFormatParameterFilter.kt

```diff
@@ -0,0 +1,77 @@
+package org.genspectrum.lapis.controller
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import jakarta.servlet.FilterChain
+import jakarta.servlet.http.HttpServletRequest
+import jakarta.servlet.http.HttpServletRequestWrapper
+import jakarta.servlet.http.HttpServletResponse
+import mu.KotlinLogging
+import org.genspectrum.lapis.util.CachedBodyHttpServletRequest
+import org.springframework.stereotype.Component
+import org.springframework.web.filter.OncePerRequestFilter
+import java.util.Collections
+import java.util.Enumeration
+
+private val log = KotlinLogging.logger {}
+
+const val TEXT_CSV_HEADER = "text/csv"
+const val TEXT_TSV_HEADER = "text/tab-separated-values"
+
+@Component
+class DataFormatParameterFilter(val objectMapper: ObjectMapper) : OncePerRequestFilter() {
+
+    override fun doFilterInternal(
+        request: HttpServletRequest,
+        response: HttpServletResponse,
+        filterChain: FilterChain,
+    ) {
+        val reReadableRequest = CachedBodyHttpServletRequest(request, objectMapper)
+
+        filterChain.doFilter(AcceptHeaderModifyingRequestWrapper(reReadableRequest), response)
+    }
+}
+
+class AcceptHeaderModifyingRequestWrapper(
+    private val reReadableRequest: CachedBodyHttpServletRequest,
+) : HttpServletRequestWrapper(reReadableRequest) {
+
+    override fun getHeader(name: String): String? {
+        if (name.equals("Accept", ignoreCase = true)) {
+            when (reReadableRequest.getRequestFields()[FORMAT_PROPERTY]?.textValue()?.uppercase()) {
+                "CSV" -> {
+                    log.debug { "Overwriting Accept header to text/csv due to format property" }
+                    return TEXT_CSV_HEADER
+                }
+
+                "TSV" -> {
+                    log.debug { "Overwriting Accept header to text/csv due to format property" }
+                    return TEXT_TSV_HEADER
+                }
+
+                else -> {}
```

Comment:
What about the case when `dataFormat=json` is set? Shouldn't it also overwrite the accept header to be consistent?

Reply:
I added it